### DISCUSSION TITLE
Please suggest a generator for all tiers or at least bases for BL and data

### DISCUSSION THREAD

**Discussion by aprokopyev** [MarimerLLC/csla#2050] [Questions]
Hello,

In the past I have made my own generator which was reading a database structure using automation of MyGeneration tool and then it generated a hierarchy of classes of 2 (+ manual inherited customization level for each automatic level )  levels of inheritance depending of an entity:

1) Automatically generated data object bases for the common library with primitives - it is for nHibernate POCOs.
1.1) Some logic was added manually in the descendants of the above classes

2) Bases for business objects based on 1.1, a few new of them were nHibernate POCOs too.
2.2) Descendants of 2) were created manually and actual business logic was written here.

I would prefer to have a ready more modern tool like MyGeneration which could generate such hierarchy of classes based on the database structure and using both CSLA for BL and nHibernate for data portal.

Can you please suggest anything like this?

Also another generator for GUI bases would be also very welcome too.

I would like to add my custom code only in the descendants of an automatically generated skeleton.
So that if I need a few more fields or tables/entities/classes/POCOs added, such change would be done in a model and all ancestors would be regenerated automatically.

**Comment by rockfordlhotka** [comment]
At one point I'd have recommended [CslaGenFork](https://github.com/MarimerLLC/CslaGenFork), and it still might be a good option. Unfortunately Tiago, who put all the work into this, went dark suddenly some time ago and I've been unable to reach or find him.

There's a list of possible generators in the docs directory: https://github.com/MarimerLLC/csla/blob/master/docs/code-generators.md
